src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -68,6 +69,30 @@ trait Reporting { this: Context =>
   def featureWarning(msg: => String, pos: SourcePosition = NoSourcePosition): Unit =
     reporter.report(new FeatureWarning(msg, pos))
 
+  private[this] var reportedFeatures = Set[Symbol]()
```

Comment:
Global vars are disallowed since they will cause races for parallel compilation.